pkg/ccl/kvccl/kvtenantccl/upgradeccl/upgradeccl_test: TestTenantAutoUpgradeNoClusterSettings failed

pkg/ccl/kvccl/kvtenantccl/upgradeccl/upgradeccl_test.TestTenantAutoUpgradeNoClusterSettings [failed](https://mesolite.cluster.engflow.com/invocations/default/6919b188-8e87-44c3-9397-90cbf0b2880e?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2t2Y2NsL2t2dGVuYW50Y2NsL3VwZ3JhZGVjY2w6dXBncmFkZWNjbF90ZXN0) on release-26.1 @ [e08ca47fea6351dbb5aa6a34c443eda6ce310b20](https://github.com/cockroachdb/cockroach/commits/e08ca47fea6351dbb5aa6a34c443eda6ce310b20):


```
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.PrefixAndTable({0x8955040, 0xc005b4b680}, {{0xc007fcca50, 0xc002ce4d80}, {{0x1, 0x0}, {0x0, 0x0, 0x0, 0x0, ...}, ...}}, ...)
	pkg/sql/catalog/descs/helpers.go:128 +0x85
github.com/cockroachdb/cockroach/pkg/sql.(*schemaResolver).LookupObject(0xc00565a110, {0x8955040, 0xc005b4b680}, {0x1, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0}, ...)
	pkg/sql/schema_resolver.go:166 +0xf8c
github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver.ResolveExisting({0x8955040, 0xc005b4b680}, 0xc006fa2c00, {0x7fe0fc16fc88, 0xc00565a110}, {0x1, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	pkg/sql/catalog/resolver/resolver.go:363 +0x30a
github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver.ResolveExistingObject({0x8955040, 0xc005b4b680}, {0x89ccf70, 0xc00565a110}, 0xc006fa2c00, {0x1, 0x0, 0x0, 0x0, 0x0, ...})
	pkg/sql/catalog/resolver/resolver.go:179 +0x16b
github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver.ResolveExistingTableObject({0x8955040, 0xc005b4b680}, {0x89ccf70, 0xc00565a110}, 0xc005b4b4b8, {0x1, 0x0, 0x0, 0x0, 0x0, ...})
	pkg/sql/catalog/resolver/resolver.go:103 +0x145
github.com/cockroachdb/cockroach/pkg/sql.(*optCatalog).ResolveDataSource(0xc005b4b4a0, {0x8955040, 0xc005b4b680}, {0x0?, 0x0?, 0x0?, 0x0?}, 0x0?)
	pkg/sql/opt_catalog.go:289 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql/opt.(*Metadata).CheckDependencies(0xc002f8f408, {0x8955040, 0xc005b4b680}, 0xc00565a6d0, {0x8a0dcb8, 0xc005b4b4a0})
	pkg/sql/opt/metadata.go:480 +0x1312
github.com/cockroachdb/cockroach/pkg/sql/opt/memo.(*Memo).IsStale(0xc002f8f408, {0x8955040, 0xc005b4b680}, 0xc00565a6d0, {0x8a0dcb8, 0xc005b4b4a0})
	pkg/sql/opt/memo/memo.go:531 +0x13e8
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc00565b090, {0x8955040, 0xc005b4b680})
	pkg/sql/plan_opt.go:910 +0x6bc
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlanInternal(0xc00565a110, {0x8955040, 0xc005b4b680})
	pkg/sql/plan_opt.go:328 +0x2e
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc00565a110, {0x8955040?, 0xc005b4b680?})
	pkg/sql/plan_opt.go:308 +0x2f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc005659908, {0x8955040, 0xc005b4b680}, 0xc00565a110)
	pkg/sql/conn_executor_exec.go:3262 +0x135
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc005659908, {0x8955040, 0xc005b4b680}, 0xc00565a110, {0x89ef050, 0xc005b4b5e0})
	pkg/sql/conn_executor_exec.go:2843 +0xa72
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc005659908, {0x8955040?, 0xc005b4b4f0?}, {{0x89806c0, 0xc005b4b450}, {0x0, 0x0, 0x0}, {0xc00302dd00, 0xf2}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1085 +0x39cc
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0x8955040?, 0xc005b4b4f0?})
	pkg/sql/conn_executor_exec.go:178 +0xaa
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xc005659908, {0x8955040, 0xc005b4b4f0}, {0x89806c0, 0xc005b4b450}, 0x0, 0xc00574f898)
	pkg/sql/conn_executor_exec.go:4501 +0x4b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc005659908, {0x8955040, 0xc005b4b4f0}, {{0x89806c0, 0xc005b4b450}, {0x0, 0x0, 0x0}, {0xc00302dd00, 0xf2}, ...}, ...)
	pkg/sql/conn_executor_exec.go:177 +0x805
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(...)
	pkg/sql/conn_executor.go:2427
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc005659908)
	pkg/sql/conn_executor.go:2432 +0x2334
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc005659908, {0x8955ed0, 0xc0063d9c20}, 0x8955040?, 0xc005b4acd0?, 0x100000001506446?)
	pkg/sql/conn_executor.go:2340 +0x1f7
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:252 +0x13a
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 2715
	pkg/sql/internal.go:249 +0x2b6

goroutine 3114 [select, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc005b7f6c0, {0x8955040, 0xc005db0050})
	GOROOT/src/database/sql/sql.go:1261 +0x87
created by database/sql.OpenDB in goroutine 18
	GOROOT/src/database/sql/sql.go:841 +0x130
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantAutoUpgradeNoClusterSettings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58720

The logs indicate that this is not a panic but a test time out due to a slow quiesce. 

```
-- Test timed out at 2026-01-15 13:41:38 UTC --
panic: test timed out after 5m0s
	running tests:
		TestTenantAutoUpgradeNoClusterSettings (5m0s)
```

However, the issue is not reproducible when I tried it locally with a count of 1000 with and without the RANDOM_SEED. Sharing below some findings that I have from analysing the logs:

From the call stack, the below goroutines seem to be relevant to the issue and are hung after a stopper quiesce during test shutdown. In both cases, the call stack involves `AdmissionControl`. I did not find any obvious issue with respect to context cancelation in either of these stack traces.

```
* INFO: slow quiesce. stack traces:

goroutine 2715 [select, 4 minutes, "cluster":"app"]:
github.com/cockroachdb/cockroach/pkg/sql.(*ieResultChannel).firstResult(0xc006fa26c0, {0x8955ed0, 0xc0063d9c20})
	pkg/sql/internal_result_channel.go:118 +0xfc
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal(_, {_, _}, {_, _}, _, _, _, {{{0x0, 0x0}}, ...}, ...)
	pkg/sql/internal.go:1390 +0xf5b
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryIteratorEx(0xc0063d9ae0, {0x8955ed0, 0xc00596f860}, {0x79a8140, 0x19}, 0xc007fcca50, {{{0x0, 0x0}}, {0x0, 0x0}, ...}, ...)
	pkg/sql/internal.go:901 +0x21e
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryIterator(0xc0063d9ae0, {0x8955ed0, 0xc00596f860}, {0x79a8140, 0x19}, 0xc007fcca50, {0xc00302dd00, 0xf2}, {0x0, 0x0, ...})
	pkg/sql/internal.go:887 +0x128
github.com/cockroachdb/cockroach/pkg/jobs.RunningJobExists({0x8955ed0, 0xc00596f860}, 0xc0059c8000, 0x0, {0x89e8e88, 0xc0063d9ae0}, {0xc003f3b48c?, 0xc001903ac0?, 0xc005684f08?})
	pkg/jobs/utils.go:65 +0x28b
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigmanager.(*Manager).createAndStartJobIfNoneExists.func1({0x8955ed0, 0xc00596f860}, {0x89e8e88, 0xc0063d9ae0})
	pkg/spanconfig/spanconfigmanager/manager.go:180 +0x8e
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn.func1({0x8955ed0?, 0xc00596f860?}, 0x89dba18?)
	pkg/sql/internal.go:1968 +0x2b
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn.func4({0x8955ed0, 0xc00596f860}, 0xc007fcca50)
	pkg/sql/internal.go:2055 +0x33e
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc007fcca50, {0x8955ed0, 0xc00596f860}, 0xc006fa2640)
	pkg/kv/txn.go:1139 +0x9e
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x8955ed0, 0xc00596f860}, 0xc007fcca50, 0x1?)
	pkg/kv/db.go:1114 +0x3e
github.com/cockroachdb/cockroach/pkg/kv.(*DB). TxnWithAdmissionControl(0xc003f3ba10?, {0x8955ed0, 0xc00596f860}, 0x7520c80?, 0x0?, 0x0, 0xc006fa2640)
	pkg/kv/db.go:1077 +0xa5
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(...)
	pkg/kv/db.go:1052
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn(0xc002e077a0, {0x8955ed0, 0xc00596f860}, 0xc001d0fc90, {0x0, 0x0, 0x4c9301?})
	pkg/sql/internal.go:2042 +0x27a
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn(0xc002e077a0, {0x8955ed0, 0xc00596f860}, 0xc002254060, {0x0, 0x0, 0x0})
	pkg/sql/internal.go:1969 +0x97
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigmanager.(*Manager).createAndStartJobIfNoneExists(0xc0081b7900, {0x8955ed0, 0xc00596f860}, 0xc0059c8000)
	pkg/spanconfig/spanconfigmanager/manager.go:179 +0x217
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigmanager.(*Manager).run.func5()
	pkg/spanconfig/spanconfigmanager/manager.go:130 +0xc5
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigmanager.(*Manager).run(0xc0081b7900, {0x8955ed0, 0xc00596f860})
	pkg/spanconfig/spanconfigmanager/manager.go:149 +0x3d7
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigmanager.(*Manager).Start.func1({0x8955ed0?, 0xc00596f860?})
	pkg/spanconfig/spanconfigmanager/manager.go:89 +0x2a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x8955ed0, 0xc00596f860})
	pkg/util/stop/stopper.go:440 +0x8a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 2236
	pkg/util/stop/stopper.go:438 +0x125
```
and 
```
goroutine 3685 [select, 4 minutes, "cluster":"app"]:
github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).Admit(0xc001e5ca00, {0x8955ed0, 0xc003bad720}, {{0x3}, 0x0, 0x188aeb031898805f, 0x0, 0x5f5e100, {0x0, 0x0, ...}})
	pkg/util/admission/work_queue.go:772 +0xf85
github.com/cockroachdb/cockroach/pkg/util/admission.(*ElasticCPUWorkQueue).Admit(0xc0018f5dc0, {0x8955ed0, 0xc003bad720}, 0x5f5e100, {{0x3}, 0x0, 0x188aeb031898805f, 0x0, 0x5f5e100, {0x0, ...}})
	pkg/util/admission/elastic_cpu_work_queue.go:83 +0xc5
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvadmission.(*controllerImpl).AdmitKVWork(0xc001735310, {0x8955ed0, 0xc003bad720}, {0x0?}, {0x0?}, 0xc0056908c0)
	pkg/kv/kvserver/kvadmission/kvadmission.go:398 +0x72b
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc00175ea88, {0x8955ed0?, 0xc003bad720?}, {0x6?}, 0xc0056908c0)
	pkg/server/node.go:1640 +0x459
github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc00175ea88, {0x8955ed0, 0xc00530a000}, 0xc0056908c0)
	pkg/server/node.go:1862 +0x27c
```

A second finding is that in the failure case, the async task "propagate-ungraceful-stop" never seems to complete (whereas every success log has the log "propagate-ungraceful-stop task terminating" followed by a tenant shutdown), suggesting an issue in tenant quiesce - though it is possible that this is a consequence of the above hung goroutines and not a separate issue.

Given that the issue is related to a timeout due to slow quiesce and not reproducible even after several repeated runs, we don't think it is a release blocker (but possibly a GA blocker) but we need more time to conclude the analysis on this and make a definitive call on this. cc: @cthumuluru-crdb

--- a/pkg/spanconfig/spanconfigmanager/manager.go
+++ b/pkg/spanconfig/spanconfigmanager/manager.go
@@ -86,6 +86,9 @@ func New(
 // recreating it if it doesn't.
 func (m *Manager) Start(ctx context.Context) error {
 	return m.stopper.RunAsyncTask(ctx, "span-config-mgr", func(ctx context.Context) {
+		ctx, cancel := m.stopper.WithCancelOnQuiesce(ctx)
+		defer cancel()
+
 		m.run(ctx)
 	})
 }

